Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts

```diff
@@ -1,65 +1,127 @@
+    /**
+     * Open Launch Test Dialog
+     * @param dialog is responsible for opening the Dialog box
+     * @param testSettingService is used to get the Url from the testsettings.service file
+     * @param route is used to take the value of Id from the active route
+     */
+    constructor(public dialog: MdDialog, private testService: TestService, private route: ActivatedRoute, private snackbarRef: MdSnackBar) {
+        this.testsettings = new Test();
+        this.validEndDate = false;
+        this.validTime = false;
+        this.validStartDate = false;
+        this.currentDate = new Date();
+        this.testNameUpdationMessage = "Test Name has been updated successfully";
+        this.settingsUpdationMessage = "The settings of the Test has been updated successfully";
+    }
 
-    //Gets the Id of the Test from the route and fills the Settings saved for the selected Test in their respective fields
+    /**
+     * Gets the Id of the Test from the route and fills the Settings saved for the selected Test in their respective fields
+     */
     ngOnInit() {
         this.testId = this.route.snapshot.params['id'];
         this.getTestSettings(this.testId);
     }
 
+    /**
+     * Open snackbar
+     * @param message contains the message to be displayed when the snackbar gets opened
+     */
+    openSnackBar(message: string) {
+        let snackBarRef = this.snackbarRef.open(message, 'Dismiss', {
+            duration: 4000,
+        });
+    }
+
     //Gets the Settings saved for a particular Test
-    getTestSettings(id:number) {
-        this.testSettingService.getSettings(id).subscribe((response) => {
+    /**
+     * Gets the Settings saved for a particular Test
+     * @param id contains the value of the Id from the route
+     */
+    getTestSettings(id: number) {
+        this.testService.getTestSettings(id).subscribe((response) => {
             this.testsettings = (response);
         });
     }
 
-    //Selects the Test Name from the text box containing it on focus 
+    /**
+     * Selects the Test Name from the text box containing it on focus
+     * @param $event is used to select the contents of the target text box
+     */
     selectAllContent($event: any) {
         $event.target.select();
-    }   
+    }
 
-    //Updates the Edited Test Name in the Database on removing focus
-    onBlurMethod(id: number, testObject: Test) {
-        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {       
+    /**
+     * Updates the Edited Test Name in the Database om removing focus
+     * @param id contains the value of the Id from the route
+     * @param testObject is an object of class Test
+     * @param value contains the Test Name of the Test which is to be edited 
+     */
+    updateTestName(id: number, testObject: Test, value: string) {
+        this.testService.updateTestSettings(id, testObject).subscribe((response) => {
```

Comment:
why at updateTestName method, you are updating test settings. ? There has to be a method which only updates the test name.